pkgs/build-support/php/build-composer-project.nix

```diff
@@ -66,13 +66,18 @@ let
         inherit composer composer-local-repo-plugin;
         inherit (finalAttrs) patches pname src vendorHash version;
 
+        php = phpDrv;
```

Comment:
That's unclear to me yet, I was awaiting from you to confirm this. I'm a bit lost with those `finalAttrs` and `previousAttrs`. Can you help?

Reply:
Yeah, it should be the default:

https://github.com/NixOS/nixpkgs/blob/70721b691ca0b76314bf9f9fee8f9ee07490a33f/pkgs/development/interpreters/php/generic.nix#L92

which sets it as the `php` attribute:
https://github.com/NixOS/nixpkgs/blob/ee5ad6173a9d9c117af6240091ac19e46789f586/pkgs/top-level/php-packages.nix#L181

and scope’s `callPackage` will pass it to `buildComposerPackage`:
https://github.com/NixOS/nixpkgs/blob/ee5ad6173a9d9c117af6240091ac19e46789f586/pkgs/top-level/php-packages.nix#L57

Reply:
OK for passing it to `buildComposerProject = callPackage ../build-support/php/build-composer-project.nix { };` Thanks!

But when it comes to `mkComposerRepository`, shouldn't it be inherited from `buildComposerProject` ?

Reply:
I think it will be, since if you override `php` in `buildComposerProject`, the `finalAttrs.php` will reflect that and you are taking `mkComposerRepository` from its scope:

https://github.com/NixOS/nixpkgs/blob/70721b691ca0b76314bf9f9fee8f9ee07490a33f/pkgs/development/interpreters/php/generic.nix#L91

https://github.com/NixOS/nixpkgs/blob/70721b691ca0b76314bf9f9fee8f9ee07490a33f/pkgs/development/interpreters/php/generic.nix#L167

Reply:
OK Good to know.